lib/ansible/modules/database/mysql/mysql_db.py

```diff
@@ -136,13 +165,16 @@
 def db_exists(cursor, db):
     res = 0
     for each_db in db:
-        res += cursor.execute("SHOW DATABASES LIKE %s", (each_db.strip().replace("_", r"\_"),))
+        res += cursor.execute("SHOW DATABASES LIKE %s", (each_db.replace("_", r"\_"),))
     return res == len(db)
 
 
 def db_delete(cursor, db):
-    query = "DROP DATABASE %s" % mysql_quote_identifier(''.join(db), 'database')
-    cursor.execute(query)
+    if not db:
+        return False
+    for each_db in db:
+        query = "DROP DATABASE %s" % mysql_quote_identifier(each_db, 'database')
```

Comment:
Here there are 2 cases,

1. If non-existing `db` is **passed** via ansible playbook then it silently passes since `existence_list` variable would be `[]` returning `False`

```
$ ansible-playbook delete_nonexisting_db.yml 
 [WARNING]: No inventory was parsed, only implicit localhost is available

 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'


PLAY [Sample] *****************************************************************************************************************************************************************************************************

TASK [Delete databases db1 and db3] *******************************************************************************************************************************************************************************
ok: [localhost]

TASK [assert successful completion of deleting existing database] *************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

TASK [run command to list databases like specified database name] *************************************************************************************************************************************************
changed: [localhost]

TASK [assert that specific databases which were deleted does not exist] *******************************************************************************************************************************************
ok: [localhost] => {
    "changed": false, 
    "msg": "All assertions passed"
}

PLAY RECAP ********************************************************************************************************************************************************************************************************
localhost                  : ok=4    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

2. If non-existing `db` is **not passed** via ansible playbook then it will throw an error
`FAILED! => {"changed": false, "msg": "error deleting database: (1008, u\"Can't drop database 'abcde'; database doesn't exist\")"}`

How do we want to have it to behave?

Reply:
I was thinking on the lines where this function could be re-used by some other scripts (and not ansible or its modules). I am unnecessarily thinking too much here and making it complex.

Short answer to the question "What happens if the database does not exist?" is that nothing is changed on the system and ansible playbook executes successfully